### PR TITLE
 🐞Bugfix: Redundant messages issue in connect

### PR DESCRIPTION
**Pull request by sarthology**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
If you open chat in 2 tabs both with different channels opened and then you message from one tab, other will show messages twice/thrice.

image
[Video](https://share.getcloudapp.com/NQug6v5G)

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

**Comment by Jessc339** [comment]
UnsubscribeUnsubscribeUnsubscribe…  Sent from Mail for Windows 10 From: Sarthak SharmaSent: Wednesday, July 22, 2020 5:47 PMTo: forem/foremCc: SubscribedSubject: [forem/forem] 🐞Bugfix: Redundant messages issue in connect (#9432) What type of PR is this? (check all applicable)[ ] Refactor[ ] Feature[X] Bug Fix[ ] Optimization[ ] Documentation UpdateDescriptionIf you open chat in 2 tabs both with different channels opened and then you message from one tab, other will show messages twice/thrice.VideoAdded tests?[ ] yes[X] no, because they aren't needed[ ] no, because I need helpAdded to documentation?[ ] docs.dev.to[ ] readme[X] no documentation neededYou can view, comment on, or merge this pull request online at:  https://github.com/forem/forem/pull/9432Commit SummaryMerge pull request #1 from thepracticaldev/masterFeature 🚀 : Ability to delete messages in chat channelsMinor Bug 🐞: Show message action only for current userTest cases addedBug 🐞: Update message id for receiverRefactoring🛠: Message controller refactoringTest Cases📝 : Specs for Delete message addedFeature 🚀 : Ability to edit messagesTest Cases📝 : Specs for Edit message addedMerge pull request #2 from SkynoxTech/feature/ability_to_delete_and_edit_messagesMerge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'origin/master'Merge conflict handledMerge remote-tracking branch 'upstream/master'merge conflicts handledMerge branch 'master' of github.com:thepracticaldev/dev.toMerge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:thepracticaldev/dev.toMerge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:thepracticaldev/dev.tomerge handledMerge branch 'master' of github.com:thepracticaldev/dev.toMerge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge remote-tracking branch 'upstream/master'Merge conflict resolvedMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:SkynoxTech/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:SkynoxTech/dev.toMerge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:SkynoxTech/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:SkynoxTech/dev.toMerge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:SkynoxTech/dev.toMerge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:SkynoxTech/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:SkynoxTech/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:SkynoxTech/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge branch 'master' of github.com:thepracticaldev/dev.toMerge remote-tracking branch 'upstream/master'Merge branch 'master' of github.com:SkynoxTech/dev.to🐞Bugfix: Redundant messages on connect channel.File ChangesM app/javascript/chat/chat.jsx (18) M db/schema.rb (1) Patch Links:https://github.com/forem/forem/pull/9432.patchhttps://github.com/forem/forem/pull/9432.diff—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

File: app/javascript/chat/chat.jsx [hunk 1]
```diff
@@ -409,7 +409,8 @@ export default class Chat extends Component {
   };
 
   removeMessage = (message) => {
-    const { activeChannelId } = this.state;
+    const { activeChannelId, messages } = this.state;
+    console.log(messages);
```
Comment (rhymes) [inline app/javascript/chat/chat.jsx:413]: don't forget to remove this

File: db/schema.rb [hunk 2]
```diff
@@ -471,7 +471,6 @@
     t.datetime "updated_at", null: false
     t.bigint "user_id"
     t.datetime "verified_at"
-    t.index ["user_id"], name: "index_email_authorizations_on_user_id"
```
Comment (rhymes) [inline db/schema.rb:474]: this has probably slipped in

**Comment by Zhao-Andy** [comment]
Looks like you have a bunch of extra commits, seems okay though?

**Pull request merged**